### ISSUE TITLE
Deploying programs fails with timeout

### ISSUE THREAD

**Issue by hathbanger** [versatus/lasr#87]
Currently, deploying a program to LASR doesn't work every time. I trigger a deploy in lasrctl which publishes it:
```
12:19:58 ~/code/versatus/versatus-javascript imports $ lasrctl deploy --build example-program -a hath -n 50cent -s 50CENT -p "50 Cent's Fungible" --initializedSupply 5000 -t 5000 -i '{"imgUrl":"https://pbs.twimg.com/media/GJ34gs4XUAAjfF1?format=jpg&name=medium","conversionRate":"1","paymentProgramAddress":"0x210dca9e7121ff926606f9715777a055fe3f1d92"}'
Using existing keypair...
Getting secret key from keypair file
Getting address from keypair file
Creating temporary test file against cli arguments...
Running test...
Validating the PROGRAM OUTPUT...
*******************************
Output is valid ✅ 
*******************************
Tested input: temp-create-input-1711954219660.json

Create method testing complete...
Publishing program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Program published.
==> cid: bafyreicgizlusq2lkipdb3l3rgxcygglptawfxe7fif2en24xjid2d4p3e
```

 but then during the registration, it times out and fails:
 ```
 Registering program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d09844176d234e73a116b6f6adcfa43e8ab5929ad994c84bc1c6d0fa33a470af" --cid "bafyreicgizlusq2lkipdb3l3rgxcygglptawfxe7fif2en24xjid2d4p3e"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d09844176d234e73a116b6f6adcfa43e8ab5929ad994c84bc1c6d0fa33a470af" --cid "bafyreicgizlusq2lkipdb3l3rgxcygglptawfxe7fif2en24xjid2d4p3e"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d09844176d234e73a116b6f6adcfa43e8ab5929ad994c84bc1c6d0fa33a470af" --cid "bafyreicgizlusq2lkipdb3l3rgxcygglptawfxe7fif2en24xjid2d4p3e"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d09844176d234e73a116b6f6adcfa43e8ab5929ad994c84bc1c6d0fa33a470af" --cid "bafyreicgizlusq2lkipdb3l3rgxcygglptawfxe7fif2en24xjid2d4p3e"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d09844176d234e73a116b6f6adcfa43e8ab5929ad994c84bc1c6d0fa33a470af" --cid "bafyreicgizlusq2lkipdb3l3rgxcygglptawfxe7fif2en24xjid2d4p3e"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
Max registration attempts reached. Aborting.
Deployment error: Error: Failed to register the program.
 ```

**Comment by hathbanger** [comment]
I went and restarted LASR. It wiped all of the state, which I was under the impression that we were writing state to EIGEN, [perhaps has something to do with this](https://github.com/versatus/lasr/issues/85). 

Just did a deployment of the VERSE token, it went really smoothly:

```
13:15:00 ~/code/versatus/versatus-javascript imports $ lasrctl deploy --build example-program -a verse -n verse -s VERSE -p "LASR protocol token" --initializedSupply 100000000 -t 100000000 -i '{"imgUrl":"https://pbs.twimg.com/profile_images/1704511091236020224/aOByHnoK_400x400.jpg","conversionRate":"1","paymentProgramAddress":"0x0000000000000000000000000000000000000000"}' -r 0x1234d4b33fa19bf655519f10e8fded9e31259b28
Using existing keypair...
Getting secret key from keypair file
Getting address from keypair file
Creating temporary test file against cli arguments...
Running test...
Validating the PROGRAM OUTPUT...
*******************************
Output is valid ✅ 
*******************************
Tested input: temp-create-input-1711957557000.json

Create method testing complete...
Publishing program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Program published.
==> cid: bafyreigzmizr522iyx3nyvi6ugoyfx7cyjy5pak53b6u7vom2pdcghygpq
Checking wallet...
Registering program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Registration successful
Program registered.
==> programAddress: 0xc8e3d7cd8a9ec4a4f0738c045af56a10f7a5a973
Creating program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Program created successfully.
==> programAddress: 0xc8e3d7cd8a9ec4a4f0738c045af56a10f7a5a973
==> symbol: VERSE
==> network: stable
==> tokenName: LASR protocol token
==> initializedSupply: 100000000
==> totalSupply: 100000000
==> recipientAddress: 0x1234d4b33fa19bf655519f10e8fded9e31259b28
======
======
======
======
>>>>>>>>>>> View Program on LASR Playground:
https://faucet.versatus.io/programs/0xc8e3d7cd8a9ec4a4f0738c045af56a10f7a5a973
          
        
```

As you can see, not even any retries were needed here.

Now during a subsequent program deploy (Faucet), I'm back hitting timeouts. I'm unsure if they're related, but figured I'd document what I'm seeing.

**Comment by hathbanger** [comment]
UPDATE! so it eventually went through!

```
13:17:10 ~/code/versatus/versatus-javascript imports $ lasrctl deploy -b example-program -a hath -n faucet -s FAUCET --p "Faucet for anything fungible" --initializedSupply 1 -t 1 -i '{"imgUrl":"https://pbs.twimg.com/profile_images/1421740863139446787/huoxhEV3_400x400.jpg"}'
Using existing keypair...
Getting secret key from keypair file
Getting address from keypair file
Creating temporary test file against cli arguments...
Running test...
Validating the PROGRAM OUTPUT...
*******************************
Output is valid ✅ 
*******************************
Tested input: temp-create-input-1711957634443.json

Create method testing complete...
Publishing program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Program published.
==> cid: bafyreibltzdpwulkq7rnsw55wz3d24p5bdk2eezqtusl364nuhk3brxppm
Checking wallet...
Registering program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d5b22f5b0318a3aa18d86f5b64dd3e835a08f4c4dc64fd4a9197a48647b11acf" --cid "bafyreibltzdpwulkq7rnsw55wz3d24p5bdk2eezqtusl364nuhk3brxppm"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d5b22f5b0318a3aa18d86f5b64dd3e835a08f4c4dc64fd4a9197a48647b11acf" --cid "bafyreibltzdpwulkq7rnsw55wz3d24p5bdk2eezqtusl364nuhk3brxppm"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d5b22f5b0318a3aa18d86f5b64dd3e835a08f4c4dc64fd4a9197a48647b11acf" --cid "bafyreibltzdpwulkq7rnsw55wz3d24p5bdk2eezqtusl364nuhk3brxppm"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Error: Failed to register program: error: Command failed: ./build/lasr_cli wallet register-program --from-secret-key --secret-key "d5b22f5b0318a3aa18d86f5b64dd3e835a08f4c4dc64fd4a9197a48647b11acf" --cid "bafyreibltzdpwulkq7rnsw55wz3d24p5bdk2eezqtusl364nuhk3brxppm"
Error: Custom { kind: Other, error: "Request timeout" }

    at registerProgram (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/cli-helpers.js:102:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.deploy [as handler] (file:///Users/andrewhathaway/code/versatus/versatus-javascript/dist/lasrctrl/commands/deploy.js:158:36)
Registration failed. Retrying...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Registration successful
Program registered.
==> programAddress: 0xebf16edf15b6da50189d3e53c7088eaf03619070
Creating program...
USING RPC URL:  http://lasr-sharks.versatus.io:9292
USING IPFS URL:  167.99.20.121:5001
Program created successfully.
==> programAddress: 0xebf16edf15b6da50189d3e53c7088eaf03619070
==> symbol: FAUCET
==> network: stable
==> tokenName: Faucet for anything fungible
==> initializedSupply: 1
==> totalSupply: 1
==> recipientAddress: 0x1234d4b33fa19bf655519f10e8fded9e31259b28
======
======
======
======
>>>>>>>>>>> View Program on LASR Playground:
https://faucet.versatus.io/programs/0xebf16edf15b6da50189d3e53c7088eaf03619070
```

**Comment by hathbanger** [comment]
no longer an issue, believe it was fixed here: #101
